fpga.chart.py

```diff
@@ -176,31 +179,28 @@ def _get_xilinx_fpga_power(self, idx):
         return self._parse_xilinx_fpgainfo(FPGA_POWER_CMD, RE_POWER_STRING)
 
 
-    def _get_fpga_temp(self, idx):
-        if os.path.exists(self.intel_cmd) and os.access(self.intel_cmd, os.X_OK):
-            self.temp[idx] = self._get_intel_fpga_temp(idx)
-            return
-        if os.path.exists(self.xilinx_cmd) and os.access(self.xilinx_cmd, os.X_OK):
-            self.temp[idx] = self._get_xilinx_fpga_temp(idx)
-            return
-
-
-    def _get_fpga_power(self, idx):
-        if os.path.exists(self.intel_cmd) and os.access(self.intel_cmd, os.X_OK):
-            self.power[idx] = self._get_intel_fpga_power(idx)
-            return
-        if os.path.exists(self.xilinx_cmd) and os.access(self.xilinx_cmd, os.X_OK):
-            self.power[idx] = self._get_xilinx_fpga_power(idx)
-            return
+    def _get_fpga_temp(self):
+        while True:
+            time.sleep(self.temp_power_update_interval - 2)
+            for idx in range(self.fpga_count):
+                if os.path.exists(self.intel_cmd) and os.access(self.intel_cmd, os.X_OK):
```

Comment:
I'd prefer checking what FPGA in `init` and then execute only one command here. Also removes the need for `continue`.